plugin/evm/imports_test.go

```diff
@@ -0,0 +1,66 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package evm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/tools/go/packages"
+)
+
+// getDependencies takes a fully qualified package name and returns a map of all
+// its recursive package imports (including itself) in the same format.
+func getDependencies(packageName string) (map[string]struct{}, error) {
+	// Configure the load mode to include dependencies
+	cfg := &packages.Config{Mode: packages.NeedDeps | packages.NeedImports | packages.NeedName | packages.NeedModule}
+	pkgs, err := packages.Load(cfg, packageName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load package %s: %v", packageName, err)
+	}
+
+	if len(pkgs) == 0 || pkgs[0].Errors != nil {
+		return nil, fmt.Errorf("failed to load package %s", packageName)
+	}
+
+	deps := make(map[string]struct{})
+	var collectDeps func(pkg *packages.Package)
+	collectDeps = func(pkg *packages.Package) {
+		if _, ok := deps[pkg.PkgPath]; ok {
+			return // Avoid re-processing the same dependency
+		}
+		deps[pkg.PkgPath] = struct{}{}
+		for _, dep := range pkg.Imports {
+			collectDeps(dep)
+		}
+	}
+
+	// Start collecting dependencies
+	collectDeps(pkgs[0])
+	return deps, nil
+}
+
+func TestMustNotImport(t *testing.T) {
+	repo := "github.com/ava-labs/coreth"
+	withRepo := func(pkg string) string {
```

Comment:
nit use const whenever possible and inline variable where it's needed

```suggestion
	withRepo := func(pkg string) string {
		const repo = "github.com/ava-labs/coreth"
```